client/src/controllers/BulkController.ts

```diff
@@ -0,0 +1,69 @@
+import { Controller } from '@hotwired/stimulus';
+
+/**
+ * Controller that adds the ability to toggle some or all the elements, ignoring the disabled ones
+ * Additionaly, it checks if all non-disabled items are selected or not. If they are, then select all
+ *
+ * @example
+ * <div data-controller="w-bulk">
+ *   <input type="checkbox" data-w-bulk-target="all">
+ *
+ *   <div>
+ *     <input type="checkbox" data-w-bulk-target="item" disabled>
+ *     <input type="checkbox" data-w-bulk-target="item">
+ *     <input type="checkbox" data-w-bulk-target="item">
+ *   </div>
+ *
+ *   <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="false">Clear all</button>
+ *   <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="true">Select all</button>
+ * </div>
+ */
+export class BulkController extends Controller {
+  /** The list of targets for this controller. */
+  static targets = ['all', 'item'];
+
+  /** The list of all select-all checkbox targets for this controller. */
+  declare readonly allTargets: HTMLInputElement[];
+
+  /** The list of all item checkbox targets for this controller. */
+  declare readonly itemTargets: HTMLInputElement[];
+
+  get activeItems() {
+    return this.itemTargets.filter(({ disabled }) => !disabled);
+  }
+
+  /**
+   * Toggles the select-all checkbox based on item checkboxes.
+   */
+  toggle(): void {
+    const isAllChecked = !this.activeItems.some((item) => !item.checked);
+
+    // Update all select all checkboxes (except disabled ones)
+    this.allTargets.forEach((target) => {
+      if (!target.disabled) {
+        // eslint-disable-next-line no-param-reassign
+        target.checked = isAllChecked;
+      }
+    });
+  }
```

Comment:
```suggestion
  /**
   * On creation, ensure that the select all checkboxes are in sync.
   */
  connect() {
    this.toggle();
  }

  /**
   * When something is toggled, ensure the select all targets are kept in sync.
   */
  toggle() {
    const isAllChecked = !this.activeItems.some((item) => !item.checked);
    this.allTargets.forEach((target) => {
      // eslint-disable-next-line no-param-reassign
      target.checked = isAllChecked;
    });
  }
```

* Add a `connect` method to sync up items, this will only be required for edge cases (such as pre-selected values if the page loads with an error, or the user goes back & forward)
* No need to NOT change the select all checkboxes if they are disabled, this is unlikely to matter and if they are disabled it would make sense to keep them in sync. Also makes your code simpler.
* No need to set `void`, it's not a huge thing but best to let TypeScript infer this.
* Remove the comment, that comment is not required as the editor will tell you what `allTargets` are as you have the JSDoc above.
* Update the description of `toggle`, the previous one looked like  a copy/paste of the toggleAll method